docs/topics/coroutines-basics.md

```diff
@@ -1,293 +1,536 @@
+Use `.launch()` to run a task alongside other work when the result isn't needed or you don't want to wait for it:
+
+```kotlin
+// Imports the kotlin.time.Duration to enable expressing duration in milliseconds
+import kotlin.time.Duration.Companion.milliseconds
+
+import kotlinx.coroutines.*
 
-suspend fun doWorld() = coroutineScope {  // this: CoroutineScope
+suspend fun main() = coroutineScope {
+    // Starts a coroutine that runs without blocking the scope
     launch {
-        delay(1000L)
-        println("World!")
+        // Delays to simulate background work
+        delay(100.milliseconds)
+        println("Sending notification in background")
     }
-    println("Hello")
+
+    // Main coroutine continues while a previous one is delayed
+    println("Scope continues")
 }
-//sampleEnd
 ```
-{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
-<!--- KNIT example-basic-03.kt -->
-> You can get the full code [here](https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/jvm/test/guide/example-basic-03.kt).
->
-{style="note"}
+{kotlin-runnable="true"}
+
+After running this example, you can see that the `main()` function isn't blocked by `.launch()` and keeps running other code while the coroutine works in the background.
+
+### .async()
+
+Use the [`.async()`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/async.html) builder function to start an asynchronous computation that runs alongside other code and returns a result you can access later.
+The result is wrapped in a [`Deferred`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-deferred/) object, which you can access by calling the `.await()` function:
+
+```kotlin
+// Imports the kotlin.time.Duration to enable expressing duration in milliseconds
+import kotlin.time.Duration.Companion.milliseconds
 
-This code also prints:
+import kotlinx.coroutines.*
 
-```text
-Hello
-World!
+suspend fun main() = withContext(Dispatchers.Default) {
+   // Starts downloading the first page
+   val firstPage = async {
+      delay(50.milliseconds)
+      "First page"
+   }
+
+   // Starts downloading the second page in parallel
+   val secondPage = async {
+      delay(100.milliseconds)
+      "Second page"
+   }
+
+   // Awaits both results and compares them
+   val pagesAreEqual = firstPage.await() == secondPage.await()
+   println("Pages are equal: $pagesAreEqual")
+}
 ```
+{kotlin-runnable="true"}
 
-<!--- TEST -->
+### runBlocking()
 
-## Scope builder and concurrency
+The [`runBlocking()`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/run-blocking.html) coroutine builder function creates coroutine scope and blocks the current [thread](#comparing-coroutines-and-jvm-threads) until
+the coroutines launched in that scope finish.
+Unlike other coroutine builders, `runBlocking()` doesn't use a shared thread pool.
```

Comment:
This is inaccurate. I'm not sure what's meant here, but `runBlocking(Dispatchers.Default)` will use a shared thread pool, whereas in `runBlocking { launch { } }`, the `launch` won't be using a shared thread pool, it will be using the thread that `runBlocking` runs on. This is fully orthogonal.